### PR TITLE
USD-6554 - add in shebang and cmd controls

### PR DESCRIPTION
**Pull request by tanant**
### Description of Change(s)
This change defines a new variable - `PYTHON_COMMAND_NAME` - which can be used to specifically target the windows .cmd wrappers, and exposes `PXR_PYTHON_SHEBANG` to allow build time override (e.g. if you're using a wrapped python instance to build, and that's meant to be build-time only or similar.)

Both will fall back to `${PYTHON_EXECUTABLE}` matching current behaviour.

This follows on from the discussion in #1438, I've just extend the closed/unmerged PR (#1445) so it's an explicit selection rather than guessing on the interpreter name (i.e. if you're doing this, you're in control of your build environment so you probably know what to specify). 

Split out to a new file instead of creating a complex arg-counting-flag-whatevering-signature for `shebang.py`

**Note:** I wasn't 100% sure what license to apply for the new `wincmd.py` file so have tagged it with the unmodified Apache 2.0 lisc. 

### Fixes Issue(s)
#1438

- [X] I have verified that all unit tests pass with the proposed changes

- [X] I have submitted a signed Contributor License Agreement

### CODE REVIEW

**Comment by tanant** [comment]
Wasn't sure the reason why the mentioned PR was closed, but saw the underlying issue was still open so have quickly drafted this up. @sunyab I think this would address your last comment there around the problem of defaulting to python3 on windows due as there's no script-level auto-guessing, it's all explicitly controlled by the user when building.

**Comment by tanant** [comment]
Wanting to check - those failing CI builds, a lot (all?) of the failures look like missing deps on the build host - is that something resulting from my PR? 

- linux build, runs fine 
- macOS build fails on missing `uic` 
- windows build is complaining about a `\h` escape, which i think is from the mixed-backslash string that cmake is being passed (`...RS;D:/USDinst/include/boost-1_70;C:\hostedtoolcache\windows\Python\3.7.9\x64\Include;CPPFILES;fi...`)
  -  EDIT: looks like will be fixed at same time #2082 is done
- failing PyPI packaging on linux is complaining about not being able to find python3

**Comment by sunyab** [comment]
Hi @tanant, we're seeing these CI failures across our builds so I don't believe this is anything specific to your change. We're looking into fixes, hopefully we can re-run the builds for this PR once we get that in. Thanks!

**Comment by sunyab** [comment]
Filed as internal issue #USD-7748

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,11 +24,21 @@ include(Packages)
...
+if (NOT PXR_PYTHON_SHEBANG)
+    set(PXR_PYTHON_SHEBANG "${PYTHON_EXECUTABLE}"
+        CACHE
+        STRING
+        "Replacement path for Python #! line."
+    )
+endif()
+
+if (NOT PYTHON_COMMAND_NAME)
+    set(PYTHON_COMMAND_NAME "${PYTHON_EXECUTABLE}"
```
Comment (tanant) [inline CMakeLists.txt]: Ah. Slight nuance/pickup by @seando-adsk - this actually _isn't_ the existing behaviour. Existing behaviour would be to use the literal string 'python'. This has the effect of putting in a hardpath to the python executable used to build this. 

TL;DR unexpected/unintended behaviour change. I'll add another commit to this branch to change that to better match 1:1

Comment (tanant) [reply CMakeLists.txt]: (aaaaand done)
